ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,44 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError && error.response.statusCode !== HttpStatus.OK) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
+      }
+      throw new Error(error)
+    }
+    return name
+  }
+
+  async putCourseName (userLoginId: string, courseId: number, newName: string): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.requestUrl<Course>(
+        `courses/${courseId}`, 'PUT', { course: { name: newName } }
+      )
+      const course = response.body
+      name = course.name
```

Comment:
@chrisrrowland what would you like to see in the response when course rename is successful? Currently this is sending the rename course name?

Reply:
Okay, what object do you want? The course object? Something like this? 
```
{
   id: XXXXX,
  name: "Some Course Name"
}
```

I'm returning the name because that provides some guarantee that the change was successful, and prevents the client from needing to fetch the updated name again using a `GET` call. Using what the client sent as the new name could obscure unforeseen transformations done by Canvas.

Reply:
Yeah I see that you're returning the name as returned by the request, I was just saying I couldn't see how the result of that transaction would be any different than the input, unless it was manipulating the input text in some way.   But I think it makes sense to return the value the way you are, since that would be safe either way.

I have gotten in the habit of returning things like this (I haven't done it in typescript so not exactly but you get the gist)
```
interface PutCourseNameResponse {
   name: string | undefined
   status: number
}
```
Frankly I've done a lot of development in isolation so I don't know how many of my personal design patterns are good or not.
I end up with a lot of different response type objects (which do accurately reflect the quantity of unique responses) which can add a bit of code..

Reply:
This is my thinking
1. If canvas is sending error messages/status code, I feel the Backend should be a passthrough in those cases
2. Some internal error (say making DB insert) failed after Canvas call, in that case the BE should wrap around the error message and status code = 500 and some custom message
3. with 200 Message I agree with @ssciolla that we don't want to send entire Object but pick what we want, and I definitely agree with @chrisrrowland that it should always be sent as Json Object. We can define case by case bases when to send the success object as is or tailor it by not sending too much information

Reply:
Okay, see latest commits, I think I have made changes incorporating all the feedback.